### PR TITLE
Add metrics: subnet, dragent_bgpspeaker_association, network_dhcpagen…

### PR DESCRIPTION
**Pull request by shhgs**

### CODE REVIEW

**Review by niedbalski — APPROVED** [review]
LGTM, thanks for the contribution!

File: exporters/neutron.go [hunk 1]
```diff
@@ -486,3 +496,59 @@ func ListSubnetsPerPool(exporter *BaseOpenStackExporter, ch chan<- prometheus.Me
...
+	dragents, err := agents.ExtractAgents(allPagesAgents)
+	if err != nil {
+		return err
+	}
+	for _, agent := range dragents {
+		pages, err := agents.ListBGPSpeakers(exporter.Client, agent.ID).AllPages()
+		if err != nil {
+			return err
+		}
+		allSpeakers, err := agents.ExtractBGPSpeakers(pages)
```
Comment (niedbalski) [inline exporters/neutron.go:547]: never shadow an error, please check this error code

**Review by niedbalski — CHANGES_REQUESTED** [review]
Please check the linting issues in your PR.

File: exporters/neutron.go (hunk 1 above)
Comment (shhgs) [reply exporters/neutron.go:547]: I've fixed it. Please check it out.

**Comment by shhgs** [comment]
I've updated the code and can you proceed with the review?

**Comment by sdmitriev1** [comment]
@niedbalski is there something that still need to be fixed? It would be really great if you could accept and merge this PR

**Review by niedbalski — DISMISSED** [review]
LGTM

**Comment by niedbalski** [comment]
@shhgs can you please review the conflicts on this PR before merging?

**Comment by shhgs** [comment]
Hi, 

In order to keep  the git history clear, I checked out the main and used

```sh
$ git cherry-pick 815acfb8bf87da39cbb2e3dd075434ea40c3e7db
$ git cherry-pick --continue
```
To resolve the conflict. 

The only file that I manually edited is `exporters/exporter_test.go`.  And you can see the modification that I made with the following command. 

```
$ git diff -b -B   75ee6889386b9f5204a3c811be28624eea97c162 815acfb8bf87da39cbb2e3dd075434ea40c3e7db -- exporters/exporter_test.go             
diff --git a/exporters/exporter_test.go b/exporters/exporter_test.go                                                                                                                                               
index 9df54908..9391af8b 100644
--- a/exporters/exporter_test.go
+++ b/exporters/exporter_test.go
@@ -90,6 +90,10 @@ var fixtures map[string]string = map[string]string{
        "/neutron/v2.0/ports":                     "neutron_ports",
        "/neutron/v2.0/network-ip-availabilities": "neutron_network_ip_availabilities",
        "/neutron/v2.0/routers":                   "neutron_routers",
+       "/neutron/v2.0/agents?agent_type=BGP+dynamic+routing+agent":                 "neutron_dragents",
+       "/neutron/v2.0/agents?agent_type=DHCP+agent":                                "neutron_dhcp_agents",
+       "/neutron/v2.0/agents/3a5b0f7e-9eff-4989-b0f7-fd39d14e6ade/bgp-drinstances": "neutron_bgp_drinstances",
+       "/neutron/v2.0/agents/840d5d68-5759-4e9e-812f-f3bd19214c7f/dhcp-networks":   "neutron_dhcp_networks",
        "/neutron/v2.0/routers/f8a44de0-fc8e-45df-93c7-f79bf3b01c95/l3-agents":      "neutron_routers_l3_agents",
        "/neutron/v2.0/routers/9daeb7dd-7e3f-4e44-8c42-c7a0e8c8a42f/l3-agents":      "neutron_routers_l3_agents",
        "/loadbalancer/v2.0/lbaas/loadbalancers":                                    "loadbalancer_loadbalancers",
```

**Comment by niedbalski** [comment]
I still see conflicts here, any chance to review them and fix?

**Pull request closed without merging**